lib/client/keyagent.go

```diff
@@ -64,6 +64,9 @@ type LocalKeyAgent struct {
 
 	// proxyHost is the proxy for the cluster that his key agent holds keys for.
 	proxyHost string
+
+	// saveKeys dictates whether the agent will save added keys to disk or not.
+	saveKeys bool
```

Comment:
This is leaking key store implementation details into the key agent. For example it does not make sense if the key store keeps the keys in memory - why not have key store handle that? Maybe even use a key store implementation which is almost FSLocalKeyStore but does not write to disk?

Reply:
Reworked the implementation to not touch `LocalKeyAgent` logic and instead work by overriding key store behaviour.

Reply:
Please a have look at the new logic which should fix this.